packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -152,6 +199,17 @@ export class Snapshot {
     return this.client.resourceSnapshot;
   }
 
+  private sortReportsByDate<T extends SnapshotReport>(report: T[]): T[] {
+    return report.sort((a, b) => b.updatedDate - a.updatedDate);
+  }
+
+  private hasBegunSynchronization(
+    _synchronizationReports?: ResourceSnapshotsSynchronizationReportModel[]
+  ): _synchronizationReports is ResourceSnapshotsSynchronizationReportModel[] {
+    const reports = this.model.synchronizationReports;
+    return Array.isArray(reports) && reports.length > 0;
```

Comment:
What else could `reports` be ?

 I'm guessing it's more of a "is not null or undefined"  check ? 

If so, just optional chaining should do the trick.

Reply:
Yeah it's only a undefined check

I tried but `reports?.length` cannot be compared to a number since it might be `undefined`
![image](https://user-images.githubusercontent.com/12199712/132529267-d01391ee-c0a6-4946-b967-b4f4beb36ff2.png)

I think the shorter alternative would be something like 
```
const reports = this.model.synchronizationReports || [];
return reports.length > 0;
```